### ISSUE TITLE
make can-observation work without cid

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-observation#29]
This can probably work with `CIDStore`.  This will mean we don't need CIDs for everything.

**Comment by justinbmeyer** [comment]
To avoid the looping problem in #30, we could use a CID like:

```
var cid = 0;
CID = function(obj){
  var cur = weakMap.get( obj )
  if( ! cur  ) {
      cid++;
      weakMap.set(obj, cid);
      cur = cid;
  }
  return cur;
}
```

This would allow us to keep using `Object` as our map.

**Comment by justinbmeyer** [comment]
closed in 4.0
